src/stan/services/pathfinder/single.hpp

```diff
@@ -0,0 +1,1201 @@
+    logger.info(debug_stream);
+  }
+}
+
+template <typename T0, typename T1, typename T2, typename T3, typename T4>
+inline void taylor_appx_full2(T0&& logger, T1&& Hk, T2&& L_hk,
+                              T3&& logdetcholHk, T4&& x_center) {
+  if (STAN_DEBUG_PATH_TAYLOR_APPX) {
+    std::cout << "---Full---\n";
+    Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, 0, ", ", ", ", "\n",
+                                 "", "", " ");
+
+    std::stringstream debug_stream;
+    debug_stream << "Hk: " << Hk.format(CommaInitFmt) << "\n";
+    debug_stream << "L_approx: \n" << L_hk.format(CommaInitFmt) << "\n";
+    debug_stream << "logdetcholHk: \n" << logdetcholHk << "\n";
+    debug_stream << "x_center: \n" << x_center.format(CommaInitFmt) << "\n";
+    logger.info(debug_stream);
+  }
+}
+
+template <typename T0, typename T1>
+inline void taylor_appx_sparse1(T0&& logger, T1&& Wkbart) {
+  if (STAN_DEBUG_PATH_TAYLOR_APPX) {
+    Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, 0, ", ", ", ", "\n",
+                                 "", "", " ");
+    std::stringstream debug_stream;
+    debug_stream << "---Sparse---\n";
+    debug_stream << "Wkbar: \n" << Wkbart.format(CommaInitFmt) << "\n";
+    logger.info(debug_stream);
+  }
+}
+
+template <typename T0, typename T1, typename T2, typename T3, typename T4,
+          typename T5, typename T6, typename T7, typename T8, typename T9,
+          typename T10, typename T11>
+inline void taylor_appx_sparse2(T0&& logger, T1&& qr, T2&& alpha, T3&& Qk,
+                                T4&& L_approx, T5&& logdetcholHk, T6&& Mkbar,
+                                T7&& Wkbart, T8&& x_center, T9&& ninvRST,
+                                T10&& ninvRSTg, T11&& Rkbar) {
+  if (STAN_DEBUG_PATH_TAYLOR_APPX) {
+    Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, 0, ", ", ", ", "\n",
+                                 "", "", " ");
+    std::stringstream debug_stream;
+    debug_stream << "Full QR: \n" << qr.matrixQR().format(CommaInitFmt) << "\n";
+    debug_stream << "Alpha: \n" << alpha.format(CommaInitFmt) << "\n";
+    debug_stream << "Qk: \n" << Qk.format(CommaInitFmt) << "\n";
+    debug_stream << "L_approx: \n" << L_approx.format(CommaInitFmt) << "\n";
+    debug_stream << "logdetcholHk: \n" << logdetcholHk << "\n";
+    debug_stream << "Mkbar: \n" << Mkbar.format(CommaInitFmt) << "\n";
+    debug_stream << "Decomp Wkbar: \n" << Wkbart.format(CommaInitFmt) << "\n";
+    debug_stream << "x_center: \n" << x_center.format(CommaInitFmt) << "\n";
+    debug_stream << "NinvRST: " << ninvRST.format(CommaInitFmt) << "\n";
+    debug_stream << "ninvRSTg: \n" << ninvRSTg.format(CommaInitFmt) << "\n";
+    debug_stream << "Rkbar: " << Rkbar.format(CommaInitFmt) << "\n";
+    logger.info(debug_stream);
+  }
+}
+}  // namespace debug
+template <typename T1, typename T2>
+inline auto crossprod(T1&& x, T2&& y) {
+  return x.transpose() * y;
+}
+
+template <typename T1>
```

Comment:
Yes I just wrote these one lines because the line is long. I'm fine with adding docs for these one line functions but I think the doc is not going to be very useful for these. Or I can get rid of them and just inline them

Reply:
you can use your judgement here---I thought we had a coding standard that required all non-internal functions to be documented because they're things we're planning to keep supporting.